src/hotspot/share/gc/shared/collectedHeap.inline.hpp

```diff
@@ -46,4 +50,19 @@ inline oop CollectedHeap::class_allocate(Klass* klass, size_t size, TRAPS) {
   return allocator.allocate();
 }
 
+#if TASKQUEUE_STATS
+template <class T, MEMFLAGS F>
+inline void CollectedHeap::print_and_reset_taskqueue_stats(GenericTaskQueueSet<T, F>* queue_set, const char* label) const {
```

Comment:
Can we please move this function to `GenericTaskQueueSet`? It seems to have little to do with `CollectedHeap`.

Reply:
That is fine with me; the reason I put it there has been that it sets up the logging specifically for GC which I did feel not appropriate for the taskqueue. Passing an outputstream to this would require to add per-collector helper methods again...

Reply:
And e.g. for ZGC for the stripe queue logging you might want a different log target...

Reply:
I'm not sure I follow. Can't we just move the whole function as is? I'm thinking all collectors want to log in the same way, so no need to separate out the logging setup.

Reply:
> [...] I'm thinking all collectors want to log in the same way [...]

That's why I initially put it in `CollectedHeap`, because the _collectors_ should determine where to log to, not necessarily the class. `CollectedHeap` already contains quite a few `print*` methods too.
There also were some experimental uses of `GenericTaskQueue` for other purposes than keeping oops I them recently (still in GC code).

Currently testing a change that simply moves the method as suggested. I do not have a very strong opinion about this.